File: C++/longestCommonPrefix.cpp
#include<iostream>
#include<vector>
#include<string>

class Solution{
    public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
        int n =strs.size();
        if(n==0){
            return "";
        }
        std::string commonPrefix=strs[0];
        for(int i =1;i<n;i++){
            int size = 0;
            while(size<commonPrefix.size() && size<strs[i].size() &&commonPrefix[size]==strs[i][size]){
                size++;
            }
            commonPrefix=commonPrefix.substr(0,size);
        }
        return commonPrefix;
    }
};

int main(){
    std::vector<std::string> str1={"abca","abc","abca","abc","abcc"};
    std::vector<std::string> str2={"dbcefg","admefg"};
    Solution solution;
    std::cout<<"\""<<solution.longestCommonPrefix(str1)<<"\""<<std::endl;
    std::cout<<"\""<<solution.longestCommonPrefix(str2)<<"\""<<std::endl;
    return 0;
}

File: python/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self , strs: list[str]) -> str:
        n=len(strs)
        if n==0:
            return ""
        common_prefix=strs[0]
        for i in range(1,n):
            size=0
            while size<len(common_prefix) and size<len(strs[i]) and common_prefix[size] == strs[i][size]:
                size +=1
            common_prefix=common_prefix[:size]
        return common_prefix
        
s1=["abca","abc","abca","abc","abcc"]
s=Solution()
print(s.longestCommonPrefix(s1))